chromium_src/chrome/browser/chrome_browser_main.cc

```diff
@@ -6,8 +6,143 @@
 #include "chrome/browser/chrome_browser_main.h"
 #include "brave/browser/brave_browser_process_impl.h"
 
+#include "base/command_line.h"
+#include "base/containers/fixed_flat_map.h"
+#include "base/strings/string_number_conversions.h"
+#include "base/strings/string_split.h"
+#include "build/build_config.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/themes/theme_service.h"
+#include "chrome/browser/themes/theme_service_factory.h"
+#include "chrome/common/chrome_switches.h"
+#include "ui/base/mojom/themes.mojom.h"
+
+namespace {
+
+class ThemeCommandLineHandler {
+ public:
+  static void ProcessThemeCommandLineSwitches(
+      const base::CommandLine* command_line,
+      ThemeService* theme_service) {
+#if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_LINUX) || \
+    BUILDFLAG(IS_CHROMEOS)
+    if (!command_line || !theme_service) {
+      return;
+    }
+
+    if (command_line->HasSwitch(switches::kSetDefaultTheme)) {
+      theme_service->UseDefaultTheme();
+      return;
+    }
+
+    if (command_line->HasSwitch(switches::kSetUserColor)) {
+      std::string value =
+          command_line->GetSwitchValueASCII(switches::kSetUserColor);
+      std::vector<std::string> rgb = base::SplitString(
+          value, ",", base::TRIM_WHITESPACE, base::SPLIT_WANT_NONEMPTY);
+      if (rgb.size() == 3) {
+        int r, g, b;
+        if (base::StringToInt(rgb[0], &r) && base::StringToInt(rgb[1], &g) &&
+            base::StringToInt(rgb[2], &b) && r >= 0 && r <= 255 && g >= 0 &&
+            g <= 255 && b >= 0 && b <= 255) {
```

Comment:
Im not sure if would look better or not, but it would be better if we were checking these value boundaries with `base::CheckedNumeric<uint8_t>(src).AssignIfValid(&value)`.